### PR TITLE
fix(cli): make apiKey optional when env var set in TypeScript client

### PR DESCRIPTION
**Pull request by petesiki**
## Description
makes apiKey an optional parameter when envVar is set

## Changes Made
- Modified `generateOptionsInterface` to add additional condition `header.headerEnvVar != null;` to determine if field should be optional or not
- added logic to initialize with `{}` if every `property.hasQuestionToken`

## Testing
tested by running:
`pnpm seed run --generator ts-sdk --path /Users/petesiki/fern/client-repos/scrapybara-docs/fern --audience external`
- wrote unit tests to confirm behavior in generated sdk

- [ ] Unit tests added/updated
- [x] Manual testing completed

### CODE REVIEW

**Comment by dsinghvi** [comment]
@petesiki https://github.com/fern-api/fern/blob/main/CONTRIBUTING.md#generator-testing is the guide to testing generator changes. 

For example, to test your change you can run 
```
pnpm seed:build
pnpm seed test --generator ts-sdk --fixture auth-environment-variables
```

File: generators/typescript/sdk/client-class-generator/src/GeneratedSdkClientClassImpl.ts [hunk 1]
```diff
@@ -1042,7 +1042,10 @@ export class GeneratedSdkClientClassImpl implements GeneratedSdkClientClass {
     public getOptionsPropertiesForSnippet(context: SdkContext): ts.ObjectLiteralElementLike[] {
         const properties: ts.ObjectLiteralElementLike[] = [];
 
-        if (!this.requireDefaultEnvironment && context.ir.environments?.defaultEnvironment == null) {
+        if (
+            !this.requireDefaultEnvironment &&
+            context.environments.getGeneratedEnvironments().hasDefaultEnvironment()
+        ) {
```
Comment (dsinghvi) [inline generators/typescript/sdk/client-class-generator/src/GeneratedSdkClientClassImpl.ts:1048]: @petesiki curious what led you to this being the fix?

Comment (dsinghvi) [reply generators/typescript/sdk/client-class-generator/src/GeneratedSdkClientClassImpl.ts:1048]: (i would look here https://github.com/fern-api/fern/blob/eff31d23aaa180caee48d55e056c3ca703b41c55/generators/typescript/sdk/client-class-generator/src/GeneratedSdkClientClassImpl.ts#L1282)

Comment (petesiki) [reply generators/typescript/sdk/client-class-generator/src/GeneratedSdkClientClassImpl.ts]: thanks @dsinghvi

**Review by dsinghvi — APPROVED** [review]

**Pull request merged**